addons/BlocklyCodeEditor/test/stateTests.js

```diff
@@ -1,44 +1,51 @@
-var setUpFunction = function () {
-    this.presenter = AddonBlocklyCodeEditor_create();
-    this.blocklyStateValue = JSON.stringify({"code":"<xml xmlns=\"http://www.w3.org/1999/xhtml\"><variables><variable type=\"\" id=\"lz_.)P~fO=i`(9z}tgOk\">item</variable></variables><block type=\"text_prompt_ext\" id=\"mFW}==osP6Y*]az/49e-\" x=\"51\" y=\"153\"><mutation type=\"TEXT\"></mutation><field name=\"TYPE\">TEXT</field></block><block type=\"text_prompt_ext\" id=\")Lj~:VnU]2*[`|PbMfQ%\" x=\"131\" y=\"219\"><mutation type=\"TEXT\"></mutation><field name=\"TYPE\">TEXT</field><value name=\"TEXT\"><block type=\"math_single\" id=\"I@js!Bu?k7i1KWD5?^:T\"><field name=\"OP\">ROOT</field><value name=\"NUM\"><block type=\"math_number\" id=\"d$sG;0/0;*22[R!@Y`+%\"><field name=\"NUM\">0</field></block></value></block></value></block><block type=\"variables_set\" id=\".HdMVm$9ht$d|RcviYIG\" x=\"144\" y=\"268\"><field name=\"VAR\" id=\"lz_.)P~fO=i`(9z}tgOk\" variabletype=\"\">item</field></block></xml>","isVisible":true});
-    this.presenter.configuration.workspace = Blockly.inject("body", {
-        toolbox: "<xml></xml>",
-        sounds: false,
-        maxBlocks: 0
-    });
+TestCase('[Blockly_Code_Editor] handling state tests', {
+    setUp: function () {
+        this.presenter = AddonBlocklyCodeEditor_create();
+        this.presenter.$view = $(document.createElement('div'));
+        Blockly.Xml = {
+            textToDom: sinon.mock(),
+            domToWorkspace: sinon.mock(),
+            workspaceToDom: sinon.mock(),
+            domToText: sinon.mock()
+        };
+    },
 
-    var that = this;
-    this.stubs = {
-        setVisibility: sinon.stub(this.presenter, 'setVisibility', function (isVisible) {
-            that.presenter.configuration.isVisible = isVisible;
-        })
-    };
-};
+    'test given state when setState was called then should update visiblity and Blockly XML': function () {
+        var mockWorkspace = {
+            'body': {
+                'toolbox': '<xml></xml>',
+                'sounds': false,
+                'maxBlocks': 0
+            }
+        };
 
-var tearDownFunction = function () {
-    this.presenter.setVisibility.restore();
-};
+        this.presenter.configuration.workspace = mockWorkspace;
```

Comment:
W tym teście chyba można zrobić od razu przypisanie tak, jak zrobiłeś to w funkcji setUp w pliku commandsTests.js (bez zmiennej mockWorkspace)

Reply:
Moim zdaniem może być i tak i tak. Czasem warto coś zapisać bardziej rozwlekle jeśli będzie w ten sposób bardziej czytelne. Pozostawiam w gestii programisty, czy chce to zmienić czy nie.